lib/std/json/write_stream.zig

```diff
@@ -168,8 +168,11 @@ pub fn WriteStream(comptime OutStream: type, comptime max_depth: usize) type {
                         return;
                     }
                 },
-                .Float => if (@floatCast(f64, value) == value) {
-                    try self.stream.print("{}", .{value});
+                .ComptimeInt => {
+                    return self.emitNumber(@as(std.math.IntFittingRange(value, value), value));
```

Comment:
This needlessly limits to what can fit in an `f64`. Better to use: `try self.stringify(value); self.popState()`
You can combine the `.ComptimeFloat` case with this too.

Reply:
nevermind, I see that this is the intention for `emitNumber`. Seems a bit awkward, but go ahead and merge I guess